src/PathDrawing/PathDrawingController.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright eeGeo Ltd (2012-2015), All Rights Reserved
+
+#include "PathDrawingController.h"
+#include "RouteBuilder.h"
+#include "VectorMath.h"
+#include "RouteStyle.h"
+#include "WayPointsFactory.h"
+#include "WayPointsRepository.h"
+#include "WayPointType.h"
+#include "LatLongAltitude.h"
+
+using namespace Eegeo;
+using namespace Eegeo::Routes;
+
+namespace ExampleApp
+{
+    namespace PathDrawing
+    {
+        
+        PathDrawingController::PathDrawingController(Eegeo::Routes::RouteService& routeService
+                                                     , Eegeo::EegeoWorld& eegeoWorld
+                                                     , AppCamera::SdkModel::AppGlobeCameraWrapper& cameraWrapper
+                                                     , PathDrawing::SdkModel::IWayPointsRepository& wayPointsRepository
+                                                     , ExampleAppMessaging::TMessageBus& messageBus)
+        : m_routeService(routeService)
+        , m_world(eegeoWorld)
+        , m_cameraWrapper(cameraWrapper)
+        , m_pWayPointsRepository(wayPointsRepository)
+        , m_createdRoutes(false)
+        , m_messageBus(messageBus)
+        , m_directionResultReceivedHandler(this, &PathDrawingController::OnSearchQueryResponseReceivedMessage)
+        , m_directionsMenuStateChangedCallback(this, &PathDrawingController::OnDirectionsMenuStateChanged)
+        {
+            m_routeThicknessPolicy.SetScaleFactor(1.0f);
+            
+            m_messageBus.SubscribeUi(m_directionResultReceivedHandler);
+            m_messageBus.SubscribeUi(m_directionsMenuStateChangedCallback);
+        }
+        
+        PathDrawingController::~PathDrawingController()
+        {
+            m_messageBus.UnsubscribeUi(m_directionResultReceivedHandler);
+            
+        }
+        
+        void PathDrawingController::Update(float dt)
+        {
+            if(m_world.Initialising())
+            {
+                return;
+            }
+
+            if(m_createdRoutes)
+            {
+                float altitude = m_cameraWrapper.GetRenderCamera().GetAltitude();
+                m_routeThicknessPolicy.SetAltitude(altitude);
+            }
+        }
+        
+        void PathDrawingController::OnSearchQueryResponseReceivedMessage(const DirectionResultSection::DirectionQueryResponseReceivedMessage& message)
+        {
+//            RemoveRoutePlan();
+            if(!m_createdRoutes)
+            {
+                CreateRoutePlan();
+            }
+        }
+        
+        void PathDrawingController::CreateRoutePlan()
+        {
+            const float routeWidth = 4.f;
```

Comment:
A lot of these look like they should be configurable options on the PathDrawingController.  Worth pulling some of these out (colours, widths, etc) to a mutable PathDrawingOptionsModel and setting them to a sensible default.